### ISSUE TITLE
Mamba.in_proj.weight.grad = None when using step() function of Mamba and do loss.backward()

### ISSUE THREAD

**Issue by BUPT-zeld151** [state-spaces/mamba#296]
We attempted to use the recurrent mode of Mamba model and got this bug. Directly using the forward() function is good. But if the code goes to the step() function, and do gradient backward to the output, the gradient of Mamba would disappear. The following figure shows the test code we used and the console. 
![mamba_bug](https://github.com/state-spaces/mamba/assets/49987126/38990e05-32f4-4935-9b9c-419f1e1d9ccf)

**Comment by tridao** [comment]
`step` is only implemented for generation, so the backward pass was not implemented.

**Comment by BUPT-zeld151** [comment]
> `step` is only implemented for generation, so the backward pass was not implemented.

Thank you for replying

**Comment by chaserd** [comment]
> > `step` is only implemented for generation, so the backward pass was not implemented.
> 
> Thank you for replying

i am Bupt student , i want talk about mamba with you.  my  wechat is 18290248811
